tests/contrib/io/matplotlib/test_simple_io.py

```diff
@@ -0,0 +1,174 @@
+
+    # generate plots
+    for index in range(5):
+        plots.append(plt.figure())
+        plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+        plt.close()
+
+    experimental_filepath = str(tmpdir.join("list_images"))
+    plot_writer = MatplotlibWriter(
+        filepath=experimental_filepath, save_args={"multiFile": True}
+    )
+    plot_writer.save(plots)
+
+    # write and compare
+    for index, plot in enumerate(plots):
+        string_index = str(index)
+        trusted_filepath = str(
+            tmpdir.join("image_we_expect_{}.png".format(string_index))
+        )
+        plot.savefig(trusted_filepath)
+
+        full_experimental_filepath = "{}/{}.png".format(
+            experimental_filepath, string_index
+        )
+        assert (
+            open(full_experimental_filepath, "rb").read()
+            == open(trusted_filepath, "rb").read()
+        )
+
+
+def test_should_write_to_image_in_dict_multi_mode(tmpdir):
+    import matplotlib.pyplot as plt
+
+    plots = dict()
+
+    # generate plots
+    for index in ["boo", "far"]:
+        filename = "{}.png".format(index)
+
+        plots[filename] = plt.figure()
+        plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+        plt.close()
+
+    directory = str(tmpdir.join("dict_images"))
+    plot_writer = MatplotlibWriter(filepath=directory, save_args={"multiFile": True})
+    plot_writer.save(plots)
+
+    # write and compare
+    for filename, plot in plots.items():
+        experimental_filepath = str(tmpdir.join("dict_images", filename))
+        trusted_filepath = str(
+            tmpdir.join("dict_images", filename.replace(".png", "_trusted.png"))
+        )
+        plot.savefig(trusted_filepath)
+
+        assert (
+            open(experimental_filepath, "rb").read()
+            == open(trusted_filepath, "rb").read()
+        )
+
+
+def test_only_lists_or_dicts_should_be_accepted_for_mulit_mode(tmpdir):
+    import matplotlib.pyplot as plt
+
+    plot_object = plt.figure()
+    plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+    plt.close()
+
+    plot_writer = MatplotlibWriter(
+        filepath=str(tmpdir.join("some_path")), save_args={"multiFile": True}
+    )
+
+    with pytest.raises(DataSetError) as excinfo:
+        plot_writer.save(plot_object)
+
+    expected_dict_error = (
+        "multiFile is True but data type not "
+        "dict or list. Rather, "
+        "<class 'matplotlib.figure.Figure'>"
+    )
+
+    assert str(excinfo.value) == expected_dict_error
+
+
+def test_load_should_fail(tmpdir):
+
+    plot_writer = MatplotlibWriter(filepath=str(tmpdir.join("some_path")))
+
+    with pytest.raises(DataSetError) as excinfo:
+        plot_writer.load()
+
+    expected_load_error = "Loading not supported for MatplotlibWriter"
+
+    assert str(excinfo.value) == expected_load_error
+
+
+def test_exists_functionality(tmpdir):
+    import matplotlib.pyplot as plt
+
+    plot_object = plt.figure()
```

Comment:
These 3 lines can be in a pytest fixture. (See [test_json_local.py](https://github.com/quantumblacklabs/kedro/blob/develop/tests/io/test_json_local.py) as an example)

Reply:
I was going to do a fixture but the implementation of the figures is slightly different and I'm not sure of the value of having something generating an object for reuse when a higher level state is concerned (.e.g. with the plt module). 

For this reason I don't think there is a value in reuse of implementing fixtures here.